src/components/DemandDetails/DemandDetails.tsx

```diff
@@ -0,0 +1,128 @@
+import { Button } from '@dataesr/react-dsfr';
+import React, { useCallback, useEffect, useState } from 'react';
+import { useServices } from 'src/services';
+import type { RowsParams } from 'src/services/demandsService';
+import {
+  demandEditableRowsParams,
+  demandRowsParams,
+} from 'src/services/demandsService';
+import { Demand } from 'src/types/Summary/Demand';
+
+const renderEditable = (
+  type: string,
+  key: keyof Demand,
+  demand: Demand,
+  onEvent?: (e: React.ChangeEvent<HTMLInputElement>) => void
+) => {
+  switch (type) {
+    case 'checkbox':
+      return (
+        <input
+          type="checkbox"
+          name={key}
+          defaultChecked={demand?.[key] as boolean}
+          onChange={onEvent}
+        />
+      );
+    case 'textarea':
+      return <textarea name={key} defaultValue={demand?.[key] as string} />;
+    default:
+      return demand?.[key] ?? 'Valeur inconnu';
+  }
+};
+
+function DemandDetails({ demandId }: { demandId: string }) {
+  const { demandsService } = useServices();
+  const [demand, setDemand] = useState<Demand | null>(null);
+
+  const updateDemand = useCallback(
+    (type: string, key: keyof Demand) =>
+      (evt: React.ChangeEvent<HTMLInputElement>) => {
+        const demandId = demand?.id;
+        const demandUpdate = {
+          [key as keyof Demand]:
+            evt.target?.[
+              ['checkbox', 'radio'].includes(type) ? 'checked' : 'value'
+            ],
+        };
+
+        if (!demandId) return;
+        demandsService
+          .updateDemand(demandId, demandUpdate)
+          .then(({ data: record }) => setDemand(record as Demand));
+      },
+    [demand, demandsService]
+  );
+
+  const handleSubmit = useCallback(
+    (evt: React.FormEvent<HTMLFormElement>) => {
+      evt.preventDefault();
+
+      const commentInput = (evt.target as Record<string, any>).elements?.[
+        'Commentaire'
+      ];
+      const demandUpdate = {
+        Commentaire: commentInput?.value || demand?.['Commentaire'] || '',
+      };
+
+      const demandId = demand?.id;
+      if (!demandId) return;
+      demandsService
+        .updateDemand(demandId, demandUpdate)
+        .then(({ data: record }) => setDemand(record as Demand));
+    },
+    [demand, demandsService]
+  );
+
+  // TODO: extract the API call to another component
```

Comment:
I don't see wy it has to be in another component ? 
but if so please create a trello ticket